docs/source/tutorials/python_in_mantid/further_alg_ws/06_extract_manipulate_data.rst

```diff
@@ -0,0 +1,146 @@
+.. _06_extract_manipulate_data:
+
+=====================================
+Extract and Manipulate Data: Examples
+=====================================
+
+
+Read vs Extract
+===============
+
+* *Read* produces a view into the chosen part of the original data.
+* *Extract* creates a copy of this part of the data.
+
+Read
+----
+
+Using a loop, read and print the first value in all spectra
+
+.. code-block:: python
+
+	for index in range(0, raw_workspace.getNumberHistograms()):
+	    #Note the round brackets followed by the square brackets
+	    print(raw_workspace.readY(index)[0])
+
+Workspace data can be read as numpy arrays, spectrum by spectrum:
+
+.. code-block:: python
+
+	ws = Load(Filename="HRP39182.RAW")
+	for i in range(ws.getNumberHistograms()):
+	    y = ws.readY(i)
+	    x = ws.readX(i)
+	    e = ws.readE(i)
+
+Warning
+-------
+**Be careful**: the outputs of *read* (y,x,e) are only **views into the data held by the workspace,** `ws`. If `ws` is deleted, the contents of x,y,e will be nonsense (the random contents of the memory locations formerly used for `ws`).
+If you need x,y,e to persist longer than ws, use the *extract*, which creates a copy of the data in `ws` into y,x,e.
+
+Extract
+-------
+
+The data from all spectra can be obtained as a mutable multi-dimensional array in one-call using the extract methods.
+
+.. code-block:: python
+
+	ws = Load(Filename="HRP39182.RAW")
+	x = ws.extractX()
+
+	print(x.shape)
+	print(y.shape)
```

Comment:
Are these print statements correct, as we don't have a `y` or `e` variable, or maybe I'm missing something.

Reply:
I'd copied that over from here and for some reason left out the extractY and extractE lines!